src/contexts/dataContext.js

```diff
@@ -1,30 +1,112 @@
-import { createContext, useContext, useEffect, useState } from "react";
-import axios from "axios";
-
-const ProductsDataContext = createContext();
-
-const ProductsDataProvider = ({ children }) => {
-  const [productsData, setProductsData] = useState([]);
-  try {
-    useEffect(
-      () =>
-        (async () => {
-          const resp = await axios.get("/api/products");
-          if (resp.status === 200) setProductsData(resp.data.products);
-        })(),
-      []
-    );
-  } catch (e) {
-    console.log("ProductsDataProvider : Error in fetching products", e);
+import { createContext, useContext, useReducer } from "react";
+import { useProductsData } from "./productContext";
+import {
+  funcFlavorCategory,
+  funcInStock,
+  funcItemCategory,
+  funcPriceRangeCategory,
+  funcRating2AndAbove,
+  funcRating3AndAbove,
+  funcRating4AndAbove,
+  funcSortHighToLow,
+  funcSortLowToHigh,
+} from "../utilities/filterFunctions";
+import { reducer } from "../reducer";
+
+const initialState = {
+  flavors: {
+    Chocolate: false,
+    Vanilla: false,
+    "Red Velvet": false,
+    Pineapple: false,
+    Strawberry: false,
+  },
+  items: {
+    Cake: false,
+    Muffin: false,
+  },
+  priceRange: {
+    under500: false,
+    price500To1000: false,
+    price1000To1500: false,
+    price1500To2000: false,
+  },
+  isOutOfStock: true,
+  sortBy: "",
+  ratingCategory: "",
+  cart: [],
+  wishlist: [],
+};
+
+const getData = (state) => {
+  let { productsData } = useProductsData();
+
+  if (state.sortBy === "highToLow")
+    productsData = funcSortHighToLow(productsData);
+  else if (state.sortBy === "lowToHigh")
+    productsData = funcSortLowToHigh(productsData);
+
+  if (!state.isOutOfStock) productsData = funcInStock(productsData);
+
+  // cakes and muffins
+
+  // getting keys with values true, keys are category names of items
+  const itemCategories = Object.keys(state.items).filter(
```

Comment:
Good to know that you have included these comments, it gives better code review

Reply:
> ![image](https://user-images.githubusercontent.com/86982322/159129953-aa6ba09d-f4f3-45b6-abfe-aa7c220e4b92.png)
> 
> Can you check out why this image is not getting load in mine? is it for me only?

Yes, the image is loading for me
![image](https://user-images.githubusercontent.com/42478246/159130660-3d78c986-3385-4a0e-acd5-668893777d00.png)